From: David Coe <david@coent.demon.co.uk>
CC: Cygnus Gnu-win32 <gnu-win32@cygnus.com>
Subject: Cygwin B16 and Sharutils 4.2

This is a multi-part message in MIME format.

--------------599619382989
Content-Type: text/plain; charset=us-ascii
Content-Transfer-Encoding: 7bit

The following should produce a working Sharutils package for NT:

1. download and unpack sharutil-4.2.tar.gz

2. apply attached patch to uuencode.c and uudecode.c

3. configure --prefix=/usr --disable-nls

4. cp intl/libgettext.h intl/libintl.h
	until ln works

5. make; make check
	should only differ on file permissions

6. make install; make man-install

7. edit /usr/man/man1/uudecode.1
	so. /usr/man/man11/uuencode.1 => so. man1/uuencode.1

8. cd /usr/bin; strip shar unshar uudecode uuencode
	add .exe to them if you want to use them outside bash

9. enjoy

-- 
Dr David Coe			     \=\
58 Fairlawn Drive, East Grinstead     \=\   Tel +44 1342 326860
West Sussex, RH19 1NT, United Kingdom  \=\  Fax +44 1342 316019

--------------599619382989
Content-Type: text/plain; charset=us-ascii; name="patch"
Content-Transfer-Encoding: 7bit
Content-Disposition: inline; filename="patch"

*** src/uuencode.c.orig	Thu Oct 03 09:39:22 1996
--- src/uuencode.c	Fri Oct 04 09:13:16 1996
***************
*** 51,56 ****
--- 51,61 ----
  
  #include "system.h"
  
+ #ifdef _WIN32
+ #include <stdio.h>
+ #include <fcntl.h>
+ #endif
+ 
  /*=======================================================\
  | uuencode [INPUT] OUTPUT				 |
  | 							 |
***************
*** 113,122 ****
--- 118,133 ----
  
  static void
  encode ()
+ 
  {
    register int ch, n;
    register char *p;
    char buf[80];
+ 
+ #ifdef _WIN32
+   setmode( fileno( stdout ), O_BINARY );
+   setmode( fileno( stdin  ), O_BINARY );
+ #endif
  
    while (1)
      {
*** src/uudecode.c.orig	Thu Oct 03 09:39:22 1996
--- src/uudecode.c	Fri Oct 04 09:30:04 1996
***************
*** 340,346 ****
--- 340,350 ----
    /* Create output file and set mode.  */
  
    if (strcmp (outname, "/dev/stdout") != 0 && strcmp (outname, "-") != 0
+ #ifdef _WIN32
+       && (freopen (outname, "wb", stdout) == NULL
+ #else
        && (freopen (outname, "w", stdout) == NULL
+ #endif
  #if HAVE_FCHMOD
  	  || fchmod (fileno (stdout), mode & (S_IRWXU | S_IRWXG | S_IRWXO))
  #else

--------------599619382989--

-
For help on using this list, send a message to
"gnu-win32-request@cygnus.com" with one line of text: "help".

